### PR TITLE
Update benchmark definitions

### PR DESCRIPTION
**Pull request by mariecwhite**
Introduces `DataType` and `MetaModel` data classes. `MetaModel` consists of higher level model information where as `Model` is an more concrete representation, consisting of inputs and outputs.

Also dumps more data to the json files.

### CODE REVIEW

File: oobi/benchmark-definitions/python/data_types.py [hunk 1]
```diff
@@ -66,6 +74,19 @@ class ModelArtifact(object):
   source_url: str
 
 
+@serialization.serializable
+@dataclass(frozen=True)
+class MetaModel(object):
```
Comment (pzread) [inline oobi/benchmark-definitions/python/data_types.py:79]: How about use `Model` for this and rename the existing `Model` to `ExecutionBenchmark`?

File: oobi/benchmark-definitions/python/data_types.py [hunk 2]
```diff
@@ -75,9 +96,7 @@ class Model(object):
   name: str
   # Tags that describe the model characteristics.
   tags: List[str]
-  framework_type: ModelFrameworkType
-  # Source of the model implementation.
-  source_info: str
+  meta_model_id: str
   input_batch_size: int
   inputs: ModelData
   outputs: ModelData
```
Comment (pzread) [inline oobi/benchmark-definitions/python/data_types.py:102]: Should `artifacts: List[ModelArtifact]` belong to `MetaModel`?

File: oobi/benchmark-definitions/python/data_types.py (hunk 1 above)
Comment (pzread) [reply oobi/benchmark-definitions/python/data_types.py:79]: Oh, I misunderstood the code. This is a higher-level abstraction for models with different batch sizes.

File: oobi/benchmark-definitions/python/data_types.py [hunk 3]
```diff
@@ -75,9 +96,7 @@ class Model(object):
   name: str
   # Tags that describe the model characteristics.
   tags: List[str]
-  framework_type: ModelFrameworkType
-  # Source of the model implementation.
-  source_info: str
+  meta_model_id: str
```
Comment (pzread) [inline oobi/benchmark-definitions/python/data_types.py:99]: Should this be `meta_model: MetaModel`?

Comment (mariecwhite) [reply oobi/benchmark-definitions/python/data_types.py:99]: I wasn't sure whether to use the id or the object. Using the object means one less lookup. I'll update to use the object.

File: .github/workflows/benchmark_xla.yml [hunk 4]
```diff
@@ -13,6 +13,7 @@ on:
     # Scheduled to run at 09:00 UTC and 21:00 UTC.
     - cron: '0 09,21 * * *'
   workflow_dispatch:
+  pull_request:
```
[the remark this answers is not in the document]
Comment (mariecwhite) [reply .github/workflows/benchmark_xla.yml]: Removed

**Comment by mariecwhite** [comment]
Merging so that I can build on top of these changes.

**Pull request merged**